### ISSUE TITLE
[java] UnusedPrivateMethod yields false positive for interface arguments

### ISSUE THREAD

**Issue by pbludov** [pmd/pmd#1181]
**Affects PMD Version:** 
6.4.0
**Rule:**
category/java/bestpractices.xml/UnusedPrivateMethod

**Code Sample demonstrating the issue:**
```
class TestClass {
  TestClass() {
    foo("");
  }
  private void foo(java.io.Serializable s) {
  }
}
```

**Running PMD through:** *CLI *
```
$ run.sh pmd -no-cache -d . -auxclasspath . -f text -R "category/java/bestpractices.xml/UnusedPrivateMethod"
TestClass.java:5:	Avoid unused private methods such as 'foo(java.io.Serializable)'.
```

**Comment by pbludov** [comment]
The bug was fixed somewhere between 6.6.0 and 6.12.0

**Comment by jsotuyod** [comment]
Thanks for the update!

**Comment by ylexus** [comment]
This issue, or a variant of it, is still happening in 6.13.0.

This is how to reproduce. All three classes below are defined in their own files, showing as one for simplicity:
```
final class MyClass {
    public Object create() {
        return method(new MyInterfaceImpl());
    }

    private Object method(MyInterface delegate) {
        return "blah" + delegate;
    }
}

public interface MyInterface {
    Object interfaceMethod(Object action);
}

final class MyInterfaceImpl implements MyInterface {
    @Override
    public Object interfaceMethod(Object command) {
        return new Object();
    }
}
```

Running PMD 

```
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd"
     version="6.13.0" timestamp="2019-04-08T17:08:31.624">
    <file name="package.MyClass.java">
        <violation beginline="8" endline="8" begincolumn="20" endcolumn="47" rule="UnusedPrivateMethod" ruleset="Best Practices"
                   package="package" class="MyClass" method="method"
                   externalInfoUrl="https://pmd.github.io/pmd-6.13.0/pmd_rules_java_bestpractices.html#unusedprivatemethod" priority="3">
            Avoid unused private methods such as 'method(MyInterface)'.
        </violation>
    </file>
</pmd>

```

Running via maven-pmd-plugin 3.11.0.
